dev/Scroller/APITests/ScrollerLayoutTests.cs

```diff
@@ -244,6 +244,157 @@ public void BasicAlignment()
+        public void ViewportHeight()
+        {
+            for (double scrollerContentHeight = 50.0; scrollerContentHeight <= 350.0; scrollerContentHeight += 300.0)
+            {
+                ViewportHeight(
+                    isScrollerParentSizeSet: false,
+                    isScrollerParentMaxSizeSet: false,
+                    scrollerVerticalAlignment: VerticalAlignment.Top,
+                    scrollerContentHeight);
+                ViewportHeight(
+                    isScrollerParentSizeSet: true,
+                    isScrollerParentMaxSizeSet: false,
+                    scrollerVerticalAlignment: VerticalAlignment.Top,
+                    scrollerContentHeight);
+                ViewportHeight(
+                    isScrollerParentSizeSet: false,
+                    isScrollerParentMaxSizeSet: true,
+                    scrollerVerticalAlignment: VerticalAlignment.Top,
+                    scrollerContentHeight);
+
+                ViewportHeight(
+                    isScrollerParentSizeSet: false,
+                    isScrollerParentMaxSizeSet: false,
+                    scrollerVerticalAlignment: VerticalAlignment.Center,
+                    scrollerContentHeight);
+                ViewportHeight(
+                    isScrollerParentSizeSet: true,
+                    isScrollerParentMaxSizeSet: false,
+                    scrollerVerticalAlignment: VerticalAlignment.Center,
+                    scrollerContentHeight);
+                ViewportHeight(
+                    isScrollerParentSizeSet: false,
+                    isScrollerParentMaxSizeSet: true,
+                    scrollerVerticalAlignment: VerticalAlignment.Center,
+                    scrollerContentHeight);
+
+                ViewportHeight(
+                    isScrollerParentSizeSet: false,
+                    isScrollerParentMaxSizeSet: false,
+                    scrollerVerticalAlignment: VerticalAlignment.Stretch,
+                    scrollerContentHeight);
+                ViewportHeight(
+                    isScrollerParentSizeSet: true,
+                    isScrollerParentMaxSizeSet: false,
+                    scrollerVerticalAlignment: VerticalAlignment.Stretch,
+                    scrollerContentHeight);
+                ViewportHeight(
+                    isScrollerParentSizeSet: false,
+                    isScrollerParentMaxSizeSet: true,
+                    scrollerVerticalAlignment: VerticalAlignment.Stretch,
+                    scrollerContentHeight);
+            }
+        }
+
+        private void ViewportHeight(
+            bool isScrollerParentSizeSet,
+            bool isScrollerParentMaxSizeSet,
+            VerticalAlignment scrollerVerticalAlignment,
+            double scrollerContentHeight)
+        {
+            Log.Comment("ViewportHeight test case - isScrollerParentSizeSet: " + isScrollerParentSizeSet + ", isScrollerParentMaxSizeSet: " + isScrollerParentMaxSizeSet +
```

Comment:
You can use [string interpolation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated) to avoid all the +'s in your string concatenation.

Reply:
(True for all the Log.Comments below too)

Reply:
Good idea. Thanks.